### DISCUSSION TITLE
Role based conditional display custom navigation items

### DISCUSSION THREAD

**Discussion by topegret** [filamentphp/filament#4249] [Help] [answered]
I am testing role based conditional show menu items in the nav bar using spatie permission , example code like below in the AppServiceProvider.php 
```

 public function boot()
    {
     .... ...
        //test code: Filament Registering custom navigation items
    if(auth()->user()->hasRole('admin'))
    {
    Filament::serving(function () {
    Filament::registerNavigationItems([
        NavigationItem::make('Analytics')
            ->url('https://filament.pirsch.io', shouldOpenInNewTab: false)
            ->icon('heroicon-o-presentation-chart-line')
            ->group('Reports')
            ->sort(3),
         ]);
      });
     }  
      ... ... 
    }
```

 I will get an error  says "Call to a member function hasRole() on null" it sounds to me that the user role does not load at this time. 

I don't see any problem to check the role in the Rerource file.  example: in a ProductResource.php 
```

class ProductResource extends Resource
{
  ... ... 
  public static function getEloquentQuery(): Builder
{
    if(auth()->user()->hasRole('super_admin'))
    {
        return Product::query();
    }
    return Product::where('user_id', auth()->user()->id);
}
 ...  ...
}


```
Appreciate any helps in advance.

**Comment by itrixinc** [comment]
Did you find a solution? 
I am stuck in a similar situation.

**Comment by ManojKiranA** [comment, marked as answer]
You can conditionally register the page in navigation using the below code
  
 ```
 protected static function shouldRegisterNavigation(): bool
  {
        return auth()->user()->hasRole('super_admin');
  }
```

But still the page will be accessible if you type the URL manuallay So
  
```
public function mount(): void
{
    parent::mount();
    abort_unless(auth()->user()->hasRole('super_admin'),403);
}
```

**Reply by topegret** [reply]
thank you . this is very helpful.

**Reply by topegret** [reply]
ha. it is in the document... 
[#](https://filamentphp.com/docs/2.x/admin/pages/getting-started#conditionally-hiding-pages-in-navigation)Conditionally hiding pages in navigation

**Reply by itrixinc** [reply]
How do you do it for a page from a plugin? Eg. Health status or Email log. Should be visible only to super-admin
